monai/inferers/splitter.py

```diff
@@ -58,142 +91,372 @@ def __call__(self, inputs: Any) -> Iterable[tuple[torch.Tensor, Sequence[int]]]:
 
 
 class SlidingWindowSplitter(Splitter):
+    """Split the input into patches with sliding window strategy and a possible overlap.
+    It also allows to offset the starting position and filter the patches.
+
+    Args:
+        patch_size : the size of the patches to be generated.
+        offset: the amount of offset for the patches with respect to the original input.  Defaults to 0.
+        overlap: the amount of overlap between patches in each dimension. It can be either a float in
+            the range of [0.0, 1.0) that defines relative overlap to the patch size, or it can be a non-negative int
+            that defines number of pixels for overlap. Defaults to 0.0.
+        filter_fn: a callable to filter patches. It should accepts exactly two parameters (patch, location), and
+            return True for a patch to keep. Defaults to no filtering.
+        pad_mode: string define the mode for `torch.nn.functional.pad`. The acceptable values are
+            `"constant"`, `"reflect"`, `"replicate"`, `"circular"` or `None`. Default to `None`.
+            If None, no padding will be applied, so it will drop the patches crossing the border of
+            the image (either when the offset is negative or the image is non-divisible by the patch_size).
+        pad_value: the value for `"constant"` padding. Defaults to 0.
+        device: the device where the patches are generated. Defaults to the device of inputs.
+
+    Note:
+        When a scaler value is provided for `patch_size`, `offset`, or `overlap`,
+            it is broadcasted to all the spatial dimensions.
+    """
+
     def __init__(
         self,
         patch_size: Sequence[int] | int,
+        overlap: Sequence[float] | float | Sequence[int] | int = 0.0,
         offset: Sequence[int] | int = 0,
-        overlap: Sequence[float] | float = 0.0,
         filter_fn: Callable | None = None,
+        pad_mode: str | None = None,
```

Comment:
I would suggest to use `PytorchPadMode.CONSTANT` as the default pad_mode, otherwise, the shape of the inferred images may not be the same as the original by default. What do you think?

Reply:
default padding is "constant" now.